idl/sprot.idol

```diff
@@ -130,6 +130,34 @@ Interface(
             ),
             encoding: Hubpack,
             idempotent: true,
-        )
+        ),
+        "caboose_size": (
+            doc: "returns the size of the caboose",
+            args: {
+                "slot": "SlotId",
+            },
+            reply: Result(
+                ok: "u32",
+                err: Complex("CabooseOrSprotError"),
+            ),
+            idempotent: true,
+            encoding: Hubpack,
+        ),
+        "read_caboose_region": (
```

Comment:
We talked about this in chat, but for future reference, I _tried_ to put a `read_caboose_value` function into the server and ran into a fun issue:

- Anything implementing `tlvc::TlvcRead` must be `Clone`
- The SP-RoT bridge uses a mutable `tx_buf: &mut u8` for writing SPI data

Together, this means that we can't implement `TlvcRead` on the `drv_stm32h7_sprot_server::ServerImpl` side of the IPC bridge.

To improve the UX, I did add a `read_caboose_value` on the _client_ side, which works because the IPC handle can be freely cloned.